File: DataStructureI/1.two-sum.cpp
#include <map>
#include <iostream>
#include <algorithm>

using namespace std;
/*
 * @lc app=leetcode id=1 lang=cpp
 *
 * [1] Two Sum
 */

// @lc code=start
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        map<int, int> maps;
        vector<int> result;
        for (int index = 0; index < nums.size(); index++){
            if (maps.count(nums[index]))
            {
                result.push_back(maps[nums[index]]);
                result.push_back(index);
                break;
            }
            else{
                maps[target - nums[index]] = index;
            }
        }
        return result;
    }
};
// @lc code=end

File: StudyPlan/ DataStructure/DataStructureI/1.two-sum/1.two-sum.py
#
# @lc app=leetcode id=1 lang=python3
#
# [1] Two Sum
#

# @lc code=start
from typing import List


class Solution:
    def twoSum(self,nums: List[int], target: int) -> List[int]:
        maps = {}
        result = []
        for index in range(len(nums)):
            complement = target - nums[index]
            if complement in maps:
                result.append(maps[complement])
                result.append(index)
                break
            else:
                maps[nums[index]] = index
        return result        
# @lc code=end

